dags/dag_content/tower_metrics_content.py

```diff
@@ -0,0 +1,411 @@
+        ExcludeCharacters="@",
+        ExcludePunctuation=True,
+        IncludeSpace=False,
+        RequireEachIncludedType=True,
+    )
+    password = response["RandomPassword"]
+    return password
+
+@task
+def modify_database_clone(aws_creds: dict, clone_name: str, password: str):
+    """
+    modifies cloned database cluster to have new master password
+
+    Args:
+        aws_creds (dict): dictionary containing aws credentials
+        clone_name (str): name of cloned database cluster
+        password (str): password generated in previous step
+    """
+    rds = create_rds_client(aws_creds)
+
+    rds.modify_db_cluster(
+        ApplyImmediately=True,
+        DBClusterIdentifier=clone_name,
+        MasterUserPassword=password,
+        EnableHttpEndpoint=True,
+    )
+    # ensure modification is complete before moving on
+    check_database_process_complete(client=rds,
+        waiter_type='db_cluster_available',
+        db_name=clone_name
+    )
+
+@task
+def update_secret(
+    aws_creds: dict, clone_name: str, db_info: dict, password: str
+) -> str:
+    """
+    updates secret in secret manager with formatted string includung new database info and random password
+
+    Args:
+        aws_creds (dict): dictionary containing aws credentials
+        clone_name (str): name of cloned database cluster
+        db_info (dict): dictionary with information from cloned database returned from clone_tower_cluster
+        password (str): randomly generated password assigned to cloned database
+
+    Returns:
+        str: secret arn string needed to access clone for queries
+    """
+    secrets = create_secret_client(aws_creds)
+
+    secret_string = (
+        "{"
+        + f'"dbInstanceIdentifier":"{clone_name}","engine":"aurora-mysql","host":"{db_info["host"]}","port":3306,"resourceId":"{db_info["resource_id"]}","username":"{db_info["user"]}","password":"{password}"'
+        + "}"
+    )
+    response = secrets.update_secret(
+        SecretId="Programmatic-DB-Clone-Access", SecretString=secret_string
+    )
+    secret_arn = response["ARN"]
+
+    return secret_arn
+
+@task
+def query_database(aws_creds: dict, resource_arn: str, secret_arn: str):
+    """
+    queries cloned database cluster with all desired queries. appends data to json_list for json export
+
+    Args:
+        aws_creds (dict): dictionary containing aws credentials
+        resource_arn (string): string containing the resource arn for the cloned database
+        secret_arn (str): string containing secret arn from update_secret task
+    """
+    rdsData = create_rds_data_client(aws_creds)
+
+    json_list = []
+
+    for query_name, query in QUERY_DICT.items():
+        response = rdsData.execute_statement(
+            resourceArn=resource_arn,
+            secretArn=secret_arn,
+            database="tower",
+            includeResultMetadata=True,
+            sql=query,
+        )
+
+        json_list.append(package_query_data(query_name, response))
```

Comment:
If you use my version of `package_query_data()` (see other comment), you'll need to update this part as follows:

```suggestion
        query_data = package_query_data(response)
        query_dict = {query_name: query_data}
        json_list.append(query_dict)
```